calico-enterprise/networking/gateway-api.mdx

```diff
@@ -0,0 +1,325 @@
+      targetPort: 80
+      protocol: TCP
+  type: ClusterIP
+  selector:
+    app: echoserver
+EOF
+```
+
+- Create a client pod that we can test from:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-client
+spec:
+  containers:
+  - args:
+    - "3600"
+    command:
+    - /bin/sleep
+    image: ubuntu
+    imagePullPolicy: Always
+    name: c1
+  terminationGracePeriodSeconds: 0
+EOF
+```
+
+- Once that pod is running, install `curl` in it:
+
+```bash
+kubectl exec -it test-client -- apt-get -y update
+kubectl exec -it test-client -- apt-get -y install curl
+```
+
+- Configure a Gateway:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: eg
+spec:
+  gatewayClassName: tigera-gateway-class
+  listeners:
+    - name: http
+      protocol: HTTP
+      port: 80
+EOF
+```
+
+- Configure routes to the two echo servers:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: ns1-2-echo
+spec:
+  parentRefs:
+    - name: eg
+  rules:
+    - backendRefs:
+        - group: ""
+          kind: Service
+          name: echoserver
+          namespace: ns1
+          port: 80
+          weight: 1
+      matches:
+        - path:
+            type: PathPrefix
+            value: /ns1
+    - backendRefs:
+        - group: ""
+          kind: Service
+          name: echoserver
+          namespace: ns2
+          port: 80
+          weight: 1
+      matches:
+        - path:
+            type: PathPrefix
+            value: /ns2
+EOF
+```
+
+- Configure the ReferenceGrants that are needed to allow the Gateway to reference services in other namespaces:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: ReferenceGrant
+metadata:
+  name: echo
+  namespace: ns1
+spec:
+  from:
+  - group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    namespace: default
+  to:
+  - group: ""
+    kind: Service
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: ReferenceGrant
+metadata:
+  name: echo
+  namespace: ns2
+spec:
+  from:
+  - group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    namespace: default
+  to:
+  - group: ""
+    kind: Service
+EOF
+```
+
+- Find the cluster IP of the Gateway Service:
+
+```bash
+kubectl get services -n tigera-gateway-system -o wide
+```
+
+- Curl from the test client pod to a URL that should be handled by the echo server in namespace `ns1`, via the Gateway:
+
+```bash
+curl http://<cluster IP>/ns1/subpath?query=demo | jq
+```
+
+- The output confirms that the request was handled by the echo server in namespace `ns1`:
+
+```
+```
+
+- Curl from the test client pod to a URL that should be handled by the echo server in namespace `ns2`, via the Gateway:
+
+```bash
+curl http://<cluster IP>/ns2/subpath?query=demo | jq
+```
+
+- The output confirms that the request was handled by the echo server in namespace `ns2`:
+
+```
```

Comment:
Is this supposed to be empty as well?

Reply:
Sorry, this was work-in-progress.  It's filled in now.